app/src/main/java/org/fossasia/openevent/app/core/event/list/EventsListAdapter.java

```diff
@@ -28,26 +28,32 @@
 class EventsListAdapter extends RecyclerView.Adapter<EventsListAdapter.EventRecyclerViewHolder>
     implements StickyRecyclerHeadersAdapter<HeaderViewHolder>, Filterable {
 
-    private final List<Event> events;
+    private List<Event> events;
     private final List<Event> selectedEvents = new ArrayList<>();
 
     private final Bus bus;
-    private final EventsPresenter eventsPresenter;
     private boolean sortByName;
 
     private boolean isLongPressed;
 
-    EventsListAdapter(List<Event> events, Bus bus, EventsPresenter eventsPresenter) {
+    private final EventListFragment fragment;
+
+    EventsListAdapter(List<Event> events, Bus bus, EventListFragment fragment) {
         this.events = events;
         this.bus = bus;
-        this.eventsPresenter = eventsPresenter;
+        this.fragment = fragment;
     }
 
     public void categorizeEvents() {
 
         notifyDataSetChanged();
     }
 
+    public void setEvents(List<Event> events) {
+        this.events = events;
+        notifyDataSetChanged();
```

Comment:
This is not an optimal way to do it. Use `DiffUtil`. For more info, check the android architecture samples.

Reply:
https://medium.com/@iammert/using-diffutil-in-android-recyclerview-bdca8e4fbb00

This is good too.